src/router/AppRouter.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from "react";
+import { Routes, Route, useNavigate, Navigate } from "react-router-dom";
+import { UserForm } from "../pages/Form";
+import { Dashboard } from "../pages/Dashboard";
+import { Layout } from "../shared/Layout";
+import { useAuth } from "../hooks/useAuth";
+import { AddVacation } from "../pages/AddVacation";
+
+const AppRouter = () => {
+  const { user, loading } = useAuth();
+  const [isBackButton, setIsBackButton] = useState<boolean>(false);
+  const [title, setTitle] = useState<string>("Dashboard");
```

Comment:
I didn't get the trick with isBackButton and title. It definitely shouldn't be here, in the Router.

It looks like Layout.tsx is just a header component?

So the idea is next: let's say you have a Dashboard component. It consists of a Header, filters/controls and table/calendar parts. The header is also used in other components. The only difference is in the title text and back-btn. 
So you need to create a stateless component Header.tsx that takes 2 props: title and backBtnAction. 

Now you can use Header in Dashboard and pass title="dashboard" without backBtnAction (which means it won't be displayed).
And in AddVacation.tsx you can use Header also. Just by passing title="Add Vacation"(or Edit, AddVacation will handle that) and backBtnAction={navigation.back()}